hydra-node/src/Hydra/Logging.hs

```diff
@@ -96,8 +96,7 @@ withTracerOutputTo ::
   (Tracer m msg -> IO a) ->
   IO a
 withTracerOutputTo hdl namespace action = do
-  hdlMutex <- newMVar ()
```

Comment:
@abailly-iohk, we share a file descriptor between several concurrent threads so, without a mutex to protect this access, I'm afraid we can't ensure that byte strings are always written as we expect them to be.

Looking at the [hPut](https://hackage.haskell.org/package/bytestring-0.11.4.0/docs/Data-ByteString.html#v:hPut) documentation I see no mention of any kind of guarantee when writing concurrently on shared file descriptor (Handle in Haskell lingo).

Of course, depending on how the Haskell thread system switching between threads, depending on how much laziness and logs frequencies we can have very diverse impact. That's the problem with undeterminism, sometimes it looks like it's working :)

But I wouldn't trust that code without this mutex, unless I've missed something obvious?

Reply:
You're not wrong.. I read hPut and it's constituents and it's just writing bytes to a file handle. No synchronization or thread-safety as far as I can tell. 

So why not go back to the original implementation?

Or instead of the empty mutex, just use a singul element queue = MVar to sequentialize the writes like before and think how the to make the last write get flushed to avoid the race condition of #690?

Reply:
I chose the simplest thing that could work, in. my opinion, to solve the mangling problem you mentioned as mutex implies less logic than queues.

Reply:
The reason why I used a queue initially was because it 1/ guarantees serialisation of writes, 2/ avoids wasting time in client threads writing to the logs thus providing more accurate timings, 3/ minimises additional synchronisation b/w threads that serialisation introduces, 4/ it's usually done that way in Haskell, with varying degrees of sophistication (check Katip's  [Scribe implementation](https://hackage.haskell.org/package/katip-0.8.7.2/docs/src/Katip.Core.html#registerScribe)).